### DISCUSSION TITLE
Use Nextras ORM?

### DISCUSSION THREAD

**Discussion by MekDrop** [ImpressCMS/impresscms#965] [Ideas]
Today I got GitHub suggesstion to look into [Nextras ORM](https://nextras.org/orm/docs/master/entity) and I think just maybe this would be much easier way to deal with many database things than writting everthing by ourselfs.

**Comment by fiammybe** [comment]
I have no issue with ORM (we are more or less doing part of ORM stuff with our criteria, but it isn't very full-featured. 
I think this will need to be postponed until we have 2.0 out and running with modules before we start working on this. There will be always one more thing to change or to improve, but inthe next version :-)

**Reply by MekDrop** [reply]
The reason why I suggested is that we have some issues with our code, why this could fix:
 * Encrypted DB columns (we need to be fully complant with technical GDPR requirements)
 * MIT and similar licensing for modules/libraries/composer - otherwise I will need probably to rewrite our ORM
 * We are lacking much needed functionality to attract developers (at least foreign keys, index'es) :(
 * Less time maintaning this functionality
 
From quick look this ORM is very close to what I did with properties and also has some functionality similar to Doctrine. So maybe all these things what we are lacking right now could be solved easy, but I don't known if with it would be possible to define properties with initVar functions (like we are using). If yes, I don't see any compatibility with older code issues. However how really that is I can't answer.

Really I would like to do something that easier would be to achieve for us good result. And if there are better ways, I would glad to look into.

**Reply by fiammybe** [reply]
As I said, I certainly have no issue with the ORM side of things, I am just afraid this will again push back a version of 2.0 that can be released. Encrypted DB columns (which are only needed if you store personal identifiable information, and just name/surname is not sensitive enough) were never part of the scope for 2.0.
There are still quite a few bugs in the current 2.0, and I'm afraid that if we start adding this, we will be losing focus.

**Reply by MekDrop** [reply]
You are right. But I'm still not sure if it would be possible to release 2.0 final faster than in 2 years with our current resources even without this functionality :|
